examples/dynamoathenamigration/migration.go

```diff
@@ -404,6 +404,10 @@ func (t *task) fromS3ToChan(ctx context.Context, dataObj dataObjectInfo, eventsC
 	afterCheckpoint := afterCheckpointIn
 
 	scanner := bufio.NewScanner(gzipReader)
```

Comment:
We could ditch the scanner and use a `json.Decoder` straight up, unless you are concerned it might blow past the 10MB? (Even so I'm not sure it's a problem, I would to some testing first.)

```go
	dec := json.NewDecoder(gzipReader)
	for dec.More() {
		var dest map[string]map[string]any
		if err := dec.Decode(&dest); err != nil {
			// ...
		}
	}
```